lib/puppet/type/scheduled_task.rb

```diff
@@ -90,6 +90,10 @@ def insync?(current)
 
     newvalue(1)
     defaultto(1)
+
+    munge do |setting|
```

Comment:
@Iristyle I haven't figured out _why_ yet but simply adding a munge here - not acting on the value at all, just returning it - is enough to prevent the weird conversion to symbol/string.

I'm going to see if I can discover anything about this behavior.

Reply:
Ok, so this ends up being really close, but we probably want to tweak it slightly. My preference would be for the former unless you have strong feelings about the alternative.

2 options:

#### Puppet 4+ solution

More strictly adhere to the value being a Puppet [number](https://puppet.com/docs/puppet/5.3/lang_data_number.html), and make the contract with users only allow `compatibility => 1` in a manifest

```ruby
validate do |value|
  raise Puppet::Error.new(_("compatibility must be a number")) unless value.is_a?(Integer)
end

# override default munging of newvalue() to symbols, treating user input as number
munge { |value| value }
```

Given validation *only* allows the given `Integer` type, and `newvalues` only allows `1`, this makes for a pretty clear enforcement.

#### Puppet 3+ solution

Accept either `compatibility => 1` OR `compatibility => '1'`

```ruby
# override default munging of newvalue() to symbols, allowing string or numeric values
munge do |value|
  if value.is_a?(String) && value =~ /^[-0-9]+$/
    Integer(value)
  else
    value
  end
end
```

FWIW, existing types and providers tend to take the latter approach, but there are a couple of reasons for this:

* It ensures that old Puppet 3 code generally still works on Puppet 4
* Nobody cared enough to make things more strict (especially in light of the first point)

I don't think either of the above points are important for net new code at this point in time. This munge also allows all types other than `String` through the `else` (like `Array`), which also makes the first variation better.